fft/src/roots_of_unity.rs

```diff
@@ -17,16 +17,28 @@ pub fn get_powers_of_primitive_root<F: IsFFTField>(
     }
 
     let root = F::get_primitive_root_of_unity(n)?;
+    let up_to = match config {
+        RootsConfig::Natural | RootsConfig::NaturalInversed => count,
+        _ => count.next_power_of_two(),
+    };
+
+    let mut results = Vec::with_capacity(up_to);
+    results.push(FieldElement::one());
+    for i in 1..up_to {
+        results.push(&root * &results[i - 1]);
+    }
 
-    //TODO: see if we can convert the successive exponentiations into an accumulated product
-    let calc = |i| match config {
-        RootsConfig::Natural | RootsConfig::NaturalInversed => root.pow(i),
-        RootsConfig::BitReverse | RootsConfig::BitReverseInversed => {
-            root.pow(reverse_index(&i, count as u64))
+    if matches!(
+        config,
+        RootsConfig::BitReverse | RootsConfig::BitReverseInversed
+    ) {
+        let mut reversed = Vec::with_capacity(count);
+        for i in 0..count {
+            reversed.push(results[reverse_index(&i, count as u64)].clone());
         }
```

Comment:
I'm curious if using `extend` would improve performance:
```suggestion
        let iter = (0..count).into_iter().map(|i| results[reverse_index(&i, count as u64)].clone());
        reversed.extend(iter);
```

Reply:
I could try that.

Reply:
It did produce a significant improvement. 10% for non-inversed and 7% for inversed.